File: src/main/java/com/devon/dapuzzo/show/ShowService.java
package com.devon.dapuzzo.show;

import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by devondapuzzo on 9/22/17.
 */
@Service
@AllArgsConstructor
public class ShowService {
    ShowRepository showRepository;

    public List<Show> getAllShows(){
        Iterable<Show> showIterable = showRepository.findAll();
        List<Show> shows = new ArrayList<>();
        showIterable.forEach(shows::add);

        return shows;
    }
}

File: src/main/java/com/devon/dapuzzo/show/ShowService.kt
package com.devon.dapuzzo.show

import org.springframework.stereotype.Service
import java.util.*

@Service
open class ShowService(val showRepository: ShowRepository) {
    fun getAllShows(): List<Show> {
            val showIterable = showRepository.findAll()
            val shows = ArrayList<Show>()
        showIterable.map { shows.add(it) }
        return shows
        }

    fun createShow(show: Show): Show {
        return Show()
    }
}
